lib/check-meta.nix

```diff
@@ -0,0 +1,200 @@
+let
+  lib = import ./default.nix;
+in rec {
+
+  # Extend a derivation with checks for brokenness, license, etc.
+  # Throw a descriptive error when the check fails; return derivaitonArg otherwise.
+  # Note: no dependencies are checked in this step.
+  addMetaChecks = { config, platform, meta, derivationArg, mkDerivationArg }: let
+    attrs = mkDerivationArg; # TODO: probably get rid of passing this one
+
+    # See discussion at https://github.com/NixOS/nixpkgs/pull/25304#issuecomment-298385426
+    # for why this defaults to false, but I (@copumpkin) want to default it to true soon.
```

Comment:
I dunno, maybe just take out the comment since I don't think people will want this turned on by default.